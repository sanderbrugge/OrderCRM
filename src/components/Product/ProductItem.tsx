import * as React from "react";
import { Product } from "../../api/Product/product";
import { View, Text } from "react-native";
import OrderButton from "../OrderButton";
import ProductStyles from "./Product.styles";
import { Item } from "../../api/Order/order";

interface IProps {
  product: Product;
  orderId: string;
  orderProduct: (orderId: string, item: Item) => void;
}

function mapProductToOrderItem(product: Product): Item {
  return {
    "product-id": product.id,
    quantity: "1",
    "unit-price": product.price,
    total: product.price
  };
}

const ProductItem: React.FC<IProps> = ({ product, orderId, orderProduct }) => {
  const item = React.useMemo(() => mapProductToOrderItem(product), [product]);
  const onOrderProductClick = React.useCallback(() => {
    orderProduct(orderId, item);
  }, [product]);

  return (
    <View style={ProductStyles.container}>
      <View style={ProductStyles.productInfoContainer}>
        <Text style={ProductStyles.productDescription}>
          {product.description}
        </Text>
        <Text style={ProductStyles.prodcutId}>{product.id}</Text>
      </View>
      <View style={ProductStyles.buttonContainer}>
        <OrderButton
          title={`$${product.price}`}
          onClick={onOrderProductClick}
        />
      </View>
    </View>
  );
};

export default ProductItem;
