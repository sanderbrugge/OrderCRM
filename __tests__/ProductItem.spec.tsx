import "react-native";
import * as React from "react";
import ProductItem from "../src/components/Product";
import renderer from "react-test-renderer";

test("ProductItem renders correctly", () => {
  const product = {
    id: "A102",
    description: "Electric screwdriver",
    category: "1",
    price: "49.50"
  };

  const orderId = "1";
  const orderProduct = jest.fn();

  const tree = renderer
    .create(
      <ProductItem
        product={product}
        orderId={orderId}
        orderProduct={orderProduct}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
