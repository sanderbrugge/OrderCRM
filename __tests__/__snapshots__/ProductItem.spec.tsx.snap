// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#a5a5a530",
      "borderBottomWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": 70,
    }
  }
>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "500",
          "marginBottom": 5,
          "marginTop": 10,
        }
      }
    >
      Electric screwdriver
    </Text>
    <Text
      style={
        Object {
          "color": "#a5a5a5",
          "fontSize": 16,
          "fontStyle": "italic",
          "fontWeight": "300",
        }
      }
    >
      A102
    </Text>
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4169E1",
          "borderRadius": 200,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#ffffff",
            "fontSize": 16,
            "justifyContent": "center",
          }
        }
      >
        <Text>
          $ 49.50
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
